Sources/SwiftParser/Expressions.swift

```diff
@@ -1394,8 +1394,21 @@ extension Parser {
     pattern: PatternContext,
     flavor: ExprFlavor
   ) -> RawMacroExpansionExprSyntax {
+    if !atStartOfFreestandingMacroExpansion() {
+      return RawMacroExpansionExprSyntax(
+        poundToken: self.consumeAnyToken(),
```

Comment:
Could you change this to call `self.expect(.pound)`? `consumeAnyToken` have historically always been sources of bugs where we could consume a token for `poundToken` that wasn’t actually a `#`.